src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/ModuleTokenResolver.cs

```diff
@@ -139,7 +139,11 @@ public void AddModuleTokenForMethod(MethodDesc method, ModuleToken token)
             {
                 MemberReference memberRef = token.MetadataReader.GetMemberReference((MemberReferenceHandle)token.Handle);
                 EntityHandle owningTypeHandle = memberRef.Parent;
-                AddModuleTokenForType(method.OwningType, new ModuleToken(token.Module, owningTypeHandle));
+                object owningType = token.Module.GetObject(owningTypeHandle, NotFoundBehavior.ReturnNull);
```

Comment:
Is `NotFoundBehavior.ReturnNull` correct here? 

When we run into missing depedencies during compilation, we should leave holes in the image. We should not generate broken code. I suspect that `NotFoundBehavior.ReturnNull` will generate broken code.

Notice that there are not existing uses of `NotFoundBehavior.ReturnNull` in this file.

Reply:
Good point, thanks, fixed in 2nd commit.